File: exercise/java/week02/exercise2/src/main/java/Book.java
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

private class Book {
    private String title;
    private String author;
    private int copies;

    public Book(String title, String author, int copies) {
        this.title = title;
        this.author = author;
        this.copies = copies;
    }

    public void addCopies(int additionalCopies) {
        if (additionalCopies > 0) {
            this.copies += additionalCopies;
        }
    }

    public void removeCopies(int soldCopies) {
        if (soldCopies > 0 && this.copies >= soldCopies) {
            this.copies -= soldCopies;
        }
    }

    public String getTitle() {
        return title;
    }

    public String getAuthor() {
        return author;
    }

    public int getCopies() {
        return copies;
    }
}

File: exercise/ts/week02-object-calisthenics/exercise2/src/book.ts
export class Book {
    private title: string;
    private author: string;
    private copies: number;

    constructor(title: string, author: string, copies: number) {
        this.title = title;
        this.author = author;
        this.copies = copies;
    }

    addCopies(additionalCopies: number): void {
        if (additionalCopies > 0) {
            this.copies += additionalCopies;
        }
    }

    removeCopies(soldCopies: number): void {
        if (soldCopies > 0 && this.copies >= soldCopies) {
            this.copies -= soldCopies;
        }
    }

    getTitle(): string {
        return this.title;
    }

    getAuthor(): string {
        return this.author;
    }

    getCopies(): number {
        return this.copies;
    }
}